test/extended/networking/cnimigration.go

```diff
@@ -0,0 +1,340 @@
+package networking
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+	networkopernames "github.com/openshift/cluster-network-operator/pkg/names"
+	"github.com/openshift/origin/test/extended/util/operator"
+
+	corev1 "k8s.io/api/core/v1"
+	eventsv1 "k8s.io/api/events/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubetypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+
+	g "github.com/onsi/ginkgo/v2"
+	"github.com/pborman/uuid"
+)
+
+const (
+	optionsEnvKey        = "TEST_SDN_LIVE_MIGRATION_OPTIONS"
+	featureGateCRName    = "cluster"
+	networkCRName        = "cluster"
+	maxMigrationDuration = 2 * time.Hour
+	// length of time to wait in-order to confirm migration is in-progress after initiating migration
+	networkOpAckTimeout = 10 * time.Minute
+)
+
+type testConfig struct {
+	targetCNI       string
+	rollbackEnabled bool
+}
+
+func getTestConfig() testConfig {
+	o := testConfig{}
+	allOptionsStr := os.Getenv(optionsEnvKey)
+	if allOptionsStr == "" {
+		return o
+	}
+	for _, opt := range strings.Split(allOptionsStr, ",") {
+		parts := strings.SplitN(opt, "=", 2)
+		if len(parts) != 2 {
+			framework.Failf("expected option of the form KEY=VALUE instead of %q", opt)
+		}
+		switch parts[0] {
+		case "target-cni":
+			o.targetCNI = parts[1]
+		case "rollback":
+			if strings.ToLower(parts[1]) == "true" {
+				o.rollbackEnabled = true
+			}
+		default:
```

Comment:
To support the usecase of https://issues.redhat.com/browse/NP-864, can we support extra options to customize the target CNI parameters in the network.operator CR?

Reply:
Done. See the options env var.

Reply:
we can add extra options yes, just add another case.